tests/rbd/rbd_trash_purge.py

```diff
@@ -0,0 +1,123 @@
+            if rbd.create_clone(snap_spec, pool, clone_name):
+                log.error(f"Clone creation failed for {clone_name}")
+
+        log.info("Moving all images created above to trash")
+        for i in range(0, num_images):
+            image_name = f"image_{i}"
+            if rbd.move_image_trash(pool, image_name):
+                log.error(f"Image move to trash failed for {image_name}")
+            if not rbd.trash_exist(pool, image_name):
+                log.error("Image is not found in the Trash")
+                raise Exception("Image is not found in the Trash")
+
+        log.info(f"Performing trash purge on pool {pool}")
+        out = rbd.trash_purge(pool)
+
+        if "some expired images could not be removed" in str(out[1]):
+            log.info(f"{str(out[1])}")
+        else:
+            log.error("rbd trash purge did not work as expected")
+
+        for i in range(0, num_images):
+            image = f"image_{i}"
+            if image in images_with_clone and not rbd.trash_exist(pool, image):
+                log.error(f"Trash purge removed image {image} which has a clone")
+                return 1
+            elif image not in images_with_clone and rbd.trash_exist(pool, image):
+                log.error(
+                    f"Image {image} is still present in trash, it should have been purged"
+                )
+                return 1
+
+        return 0
+
+    except Exception as error:
+        log.error(error.message)
+        return 1
+
+    finally:
+        rbd.clean_up(pools=[kw["config"][pool_type]["pool"]])
+
+
+def run(**kw):
+    """Verify the trash purge functionality.
+
+    This module verifies trash purge operation
+
+    Args:
+        kw: test data
+
+    Returns:
+        int: The return value. 0 for success, 1 otherwise
+
+    Pre-requisites :
+    We need atleast one client node with ceph-common package,
+    conf and keyring files
+
+    Test cases covered -
+    1) CEPH-83574482 - 'rbd trash purge' gets hung on rbds with clones or snapshots
+    and should skip to continue progress
+    Test Case Flow
+    1. Create a pool and multiple Images
+    2. generate IO in images
+    3. Create clones for few of the images
+    4. Move images to trash
+    5. Perform rbd trash purge on the pool. Purge should work without getting stuck.
+    Images without clones should get purged.
+    6. Repeat the above steps for EC-pool.
+    """
+    log.info("Running Trash image remove test")
+    rbd_obj = initial_rbd_config(**kw)
+    if rbd_obj:
+        if "rbd_reppool" in rbd_obj:
+            log.info("Ecexuting test on Replication pool")
+            if rbd_trash_purge(rbd_obj.get("rbd_reppool"), "rep_pool_config", **kw):
+                return 1
+        if "rbd_ecpool" in rbd_obj:
+            log.info("Executing test on EC pool")
+            if rbd_trash_purge(rbd_obj.get("rbd_ecpool"), "ec_pool_config", **kw):
+                return 1
```

Comment:
Optional : Induce parallelism

Reply:
Will take this as a separate task Vasi, since it requires some changes to initial_rbd_config to be able to induce parallelism